nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -281,6 +285,23 @@ public static void main(String[] args) throws IOException, InterruptedException
         }
     }
 
+    private void runDiagnostics() throws IOException {
+        final DiagnosticContext diagnosticContext = new DiagnosticContext();
```

Comment:
The `RunNiFi` class has a `bootstrapConfigFile` property referencing `bootstrap.conf`. The `DiagnosticContext` should be refactored to use the existing file reference and similar approach to loading properties as other methods in this class. Refactoring also provides an opportunity to separate configuration properties from evaluation methods.